sdks/python/apache_beam/runners/portability/artifact_service.py

```diff
@@ -57,7 +57,7 @@ def _temp_path(self, path):
     return path + '.tmp'
 
   def _open(self, path, mode):
-    return self._zipfile.open(path, mode, force_zip64=True)
+    raise NotImplementedError(type(self))
```

Comment:
FYI, this was just cleanup that I noticed.

Reply:
Unrelated, do all of the abstract methods need to be implemented in sublcasses? If yes, wondering why AbstractArtifactService is not an ABC with abstract method decorators?

Reply:
There's no reason this isn't an ABC, that would make perfect sense here.